spectator-web-spring/src/main/java/com/netflix/spectator/controllers/MetricsController.java

```diff
@@ -0,0 +1,229 @@
+ * limitations under the License.
+ */
+
+package com.netflix.spectator.controllers;
+
+import com.netflix.spectator.api.Counter;
+import com.netflix.spectator.api.DistributionSummary;
+import com.netflix.spectator.api.Gauge;
+import com.netflix.spectator.api.Id;
+import com.netflix.spectator.api.LongTaskTimer;
+import com.netflix.spectator.api.Measurement;
+import com.netflix.spectator.api.Meter;
+import com.netflix.spectator.api.Registry;
+import com.netflix.spectator.api.Timer;
+
+import com.netflix.spectator.controllers.filter.ChainedMeasurementFilter;
+import com.netflix.spectator.controllers.filter.MeasurementFilter;
+import com.netflix.spectator.controllers.filter.PrototypeMeasurementFilter;
+import com.netflix.spectator.controllers.filter.TagMeasurementFilter;
+import com.netflix.spectator.controllers.model.ApplicationRegistry;
+import com.netflix.spectator.controllers.model.MetricValues;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.io.IOException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+
+
+/**
+ * Provides an HTTP endpoint for polling spectator metrics.
+ */
+@RequestMapping("/spectator/metrics")
+@RestController
+@ConditionalOnExpression("${spectator.endpoint.enabled:false}")
+public class MetricsController {
+  @Autowired
+  private Registry registry;
+
+  @Value("${spectator.applicationName:}")
+  private String applicationName;
+
+  @Value("${spectator.applicationVersion:}")
+  private String applicationVersion;
+
+  @Value("${spectator.endpoint.tagFilter.meterNameRegex:.*}")
+  private String meterNameRegex;
+
+  @Value("${spectator.endpoint.tagFilter.tagNameRegex:}")
+  private String tagNameRegex;
+
+  @Value("${spectator.endpoint.tagFilter.tagValueRegex:}")
+  private String tagValueRegex;
+
+  @Value("${spectator.endpoint.prototypeFilter.path:}")
+  private String prototypeFilterPath;
+
+  private MeasurementFilter defaultMeasurementFilter = null;
+
+  /**
+   * The default measurement filter is configured through properties.
+   */
+  public MeasurementFilter getDefaultMeasurementFilter() throws IOException {
+    if (defaultMeasurementFilter != null) {
+      return defaultMeasurementFilter;
+    }
+
+    if (!prototypeFilterPath.isEmpty()) {
+      defaultMeasurementFilter = PrototypeMeasurementFilter.loadFromPath(
+          prototypeFilterPath);
+    } else {
+      defaultMeasurementFilter = new TagMeasurementFilter(
+          meterNameRegex, tagNameRegex, tagValueRegex);
+    }
```

Comment:
I agree with your take on the three regexes (TagMeasurementFilter), though in practice, I expect this to be useful for asking for a particular meter/measurement or closely related grouping. I think the "PrototypeMeasurementFilter" would be used specify a more complete profile using include/exclude filtering to compose comprehensive viewpoints. I have not yet actually used it in a meaningful way, but I think it addresses your remarks about specifying excludes.

The atlas filter looks interesting, but I dont want to add/support a query language at this time. I think the prototype filter allows you to express similar ideas though not as concisely.

I can take out the TagMeasurementFilter and associated query parameters if you want, perhaps adding in its place an endpoint to ask for a specific metric (but I think the query parameters are more useful). I would like to add named MeasurementFilters as a query parameter so you could poll for standard viewpoints but it seems premature to add that in now.